### DISCUSSION TITLE
[ERROR] ⚠️Imgur upload failed, check dev console (PART 2?)

### DISCUSSION THREAD

**Discussion by Trnon** [gavvvr/obsidian-imgur-plugin#64] [Getting help]
Hello, the plug-in was really great so far and I had no issues with it until today.

As I'm not really good at this programming stuff I asked around on Discord but they couldn't help me.

I looked at the Issues page and found the CORS error threads. Okay, mine is different, though. So I thought I just post it in case it actually is a different error and not the same.

Failed imgur request:  Error: {"data":{"error":"No image data was sent to the upload api","request":"\/3\/image","method":"POST"},"success":false,"status":400}
    at handleImgurErrorResponse (VM222 plugin:obsidian-imgur-plugin:506:9)
    at AuthenticatedImgurClient.upload (VM222 plugin:obsidian-imgur-plugin:541:7)
    at async ImgurAuthenticatedUploader.upload (plugin:obsidian-imgur-plugin:609:13)
    at async _ImgurPlugin.uploadFileAndEmbedImgurImage (plugin:obsidian-imgur-plugin:1089:16)
uploadFileAndEmbedImgurImage @ VM222 plugin:obsidian-imgur-plugin:1097

app.js:1 Imgur Plugin: styles.css not found

plugin:obsidian-imgur-plugin:1097 Failed imgur request:  Error: {"data":{"error":"No image data was sent to the upload api","request":"\/3\/image","method":"POST"},"success":false,"status":400}
    at handleImgurErrorResponse (plugin:obsidian-imgur-plugin:506:9)
    at AuthenticatedImgurClient.upload (plugin:obsidian-imgur-plugin:541:7)
    at async ImgurAuthenticatedUploader.upload (plugin:obsidian-imgur-plugin:609:13)
    at async _ImgurPlugin.uploadFileAndEmbedImgurImage (plugin:obsidian-imgur-plugin:1089:16)

![image](https://github.com/gavvvr/obsidian-imgur-plugin/assets/155661896/07aef661-8d64-4aad-996f-9af61cae63db)

I have no idea what all of this means so sorry if it is the CORS error and I'm sorry to have wasted your time. :/

**Comment by gavvvr** [comment]
Hi @Trnon 

I see nothing about CORS here. The error message from the API clearly says: "No image data was sent to the upload api". This is weird and sounds like there was no image payload sent by Obsidian to Imgur servers 🤔

I just checked that there are no Obsidian updates and the latest version of the plugin works. I would suggest you to try the following things to understand where is the problem:
- try creating separate clean Vault, install the plugin and check if it works there
- try copying your Vault to some other PC, open it with Obsidian (the plugin should be already there, since plugins get downloaded to `.obsidian` directory of each vault) and conduct the experiment

**Reply by Trnon** [reply]
Hi, thanks for getting back so quickly. :)

So, I've made another Vault, installed the plugin and used the anon api and it worked. I then opened the Vault that did not work, pasted the same image in and boom - uploaded. 

Idk what the issue was but it resolved itself with time? I guess...
Or maybe rebooting my PC a couple of times? 

Do you want/need anything from me to maybe troubleshoot yourself or something to recreate the issue?

If not all I want to say is thank you for the quick reply and the ongoing care for this awesome plugin. 
Have a great day.

**Reply by gavvvr** [reply]
Hi @Trnon 

Glad to hear the issue was resolved. No, I think nothing else is needed, nothing to troubleshoot on my side. 
Maybe there was something wrong with some internal Obsidian "server" which makes http requests without CORS. I have no idea how this internal component works, it's a black box for me. But I've switched to this black box from native `fetch` API to add [mobile devices support](https://github.com/gavvvr/obsidian-imgur-plugin/issues/13).

Here is that black box API for the record:
https://github.com/obsidianmd/obsidian-api/blob/bbb696aeb8bf5126bf2ecbf84fc8284fe133bc20/obsidian.d.ts#L3253C1-L3264C90

**Reply by Trnon** [reply]
I think I just figured it out.

You can open a new window on obsidian to for example work on two monitors which I do. 

I access this feature via "Open in new window". 
![image](https://github.com/gavvvr/obsidian-imgur-plugin/assets/155661896/e1baf35b-d9b8-4dc1-bc78-74fa1b1c5ba0)

Apparently the upload feature with the Imgur plugin doesn't work on the second window but on the main in does.

Is this a known issue or a new one?
